client/packages/system/src/Encounter/ListView/columns.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  useColumns,
+  ColumnAlign,
+  Column,
+  ColumnDescription,
+  SortBy,
+} from '@openmsupply-client/common';
+import { useFormatDateTime } from '@common/intl';
+import { EncounterRowFragment } from '../api';
+import { ProgramEventFragment } from '../api/operations.generated';
+
+const encounterEventCellValue = (events: ProgramEventFragment[]) => {
+  // just take the name of the first event
+  return events[0]?.name ?? '';
+};
+
+interface useEncounterListColumnsProps {
+  onChangeSortBy: (column: Column<any>) => void;
+  sortBy: SortBy<EncounterRowFragment>;
+  includePatient?: boolean;
+}
+
+export const useEncounterListColumns = ({
+  onChangeSortBy,
+  sortBy,
+  includePatient = false,
+}: useEncounterListColumnsProps) => {
+  const { localisedDate, localisedTime } = useFormatDateTime();
+  includePatient;
+
+  const columnList: ColumnDescription<EncounterRowFragment>[] = [
+    {
+      key: 'encounter-type',
+      label: 'label.encounter-type',
+      sortable: false,
+      accessor: ({ rowData }) => rowData?.document.documentRegistry?.name,
+    },
+    {
+      key: 'program',
+      label: 'label.program',
+    },
+    {
+      key: 'startDatetime',
+      label: 'label.date',
+      accessor: ({ rowData }) => rowData?.startDatetime,
+      formatter: dateString =>
+        dateString ? localisedDate((dateString as string) || '') : '',
+    },
+    {
+      key: 'startTime',
+      label: 'label.encounter-start',
+      sortable: false,
+      accessor: ({ rowData }) => rowData?.startDatetime,
+      formatter: dateString =>
+        dateString ? localisedTime((dateString as string) || '') : '',
+    },
+    {
+      key: 'endDatetime',
+      label: 'label.encounter-end',
+      formatter: dateString =>
+        dateString ? localisedTime((dateString as string) || '') : '',
+    },
+  ];
+  if (includePatient)
+    columnList.push({
+      key: 'patientId',
+      label: 'label.patient',
+      accessor: ({ rowData }) => rowData?.patient?.name,
+    });
+  columnList.push({
+    key: 'events',
+    label: 'label.label',
+    sortable: false,
+    formatter: events =>
+      encounterEventCellValue((events as ProgramEventFragment[]) ?? []),
+  });
+  columnList.push({
+    key: 'effectiveStatus',
+    label: 'label.status',
+    sortable: false,
+    align: ColumnAlign.Right,
+    width: 175,
+  });
+
+  const columns = useColumns<EncounterRowFragment>(
```

Comment:
hmm, I can see how moving ColumnDescription to a seperate file can help keep `ListView.tsx` clean, not sure about moving the useColumns declaration though, to me it's clearer to have below in `ListView.tsx` (I would say it's  a just an opinionated nit:)

```
// ListView.tsx

const columns = useColumns<EncounterRowFragment> (
   useEncounterColumnDescription(includePatient: true),
   { onChangeSortBy: updateSortQuery, sortBy }, [sortBy]
)
```

Reply:
Ohh, i see it's used in two places, never mind